src/std/misc/repr.ss

```diff
@@ -0,0 +1,133 @@
+;; -*- Gerbil -*-
+package: std/misc
+;;;; Sourceable Representation of Gerbil entities
+
+(export
+  default-representation-options current-representation-options
+  display-separated print-representation print-unrepresentable-object representable
+  pr prn repr)
+
+(import
+  :gerbil/gambit/hash :gerbil/gambit/ports
+  :std/format :std/misc/list :std/misc/rtd :std/sort :std/srfi/1)
+
+;; Default options for printing an evaluable representation. Keep it empty for now.
+;; Note: we don't actually use options yet, but
+;; it is intended to contain pretty-printing state such as desired number of columns, etc.
+;; : Table
+(def default-representation-options (make-hash-table))
+
+;; Current options for printing an evaluable representation. Always empty for now.
+;; TODO: use immutable hash tables instead when available
+;; : (Parameter Table)
+(def current-representation-options (make-parameter default-representation-options))
+
+;; Display a list of items, with prefix and suffix, separated by given separator.
+;; Print the prefix, suffix and separator strings on the given port.
+;; Call the display-element function on each element.
+;; : <- (List Any) port: Port prefix: String separator: String suffix: String display-element: (<- Any)
+(def (display-separated
+      list
+      port: (port (current-output-port))
+      prefix: (prefix "")
+      separator: (separator " ")
+      suffix: (suffix "")
+      display-element: (display-element (lambda (x) (display x port))))
```

Comment:
the `display-element` procedure should take the port as argument, since it is given to the function.

Reply:
I initially did it that way, but in practice it was a bit awkward. Oh well. Not a big deal.